@here/harp-vectortile-datasource/lib/VectorTileDecoder.ts

```diff
@@ -337,122 +291,32 @@ export class VectorTileDecoder extends ThemedTileDecoder {
-                            );
-                        });
-                    } else {
-                        featureModifiers.push(
-                            this.createFeatureModifier(filterDescription, FeatureModifierId.default)
-                        );
-                    }
-                    this.m_featureModifiers = featureModifiers;
-                } else {
-                    // null is the signal to clear the filter/modifier
-                    this.m_featureFilter = undefined;
-                    this.m_featureModifiers = undefined;
-                }
-            }
-
-            if (omvOptions.politicalView !== undefined) {
-                const politicalView = omvOptions.politicalView;
-                let featureModifiers = this.m_featureModifiers;
-                // Remove existing political view modifiers, this actually setups default,
-                // commonly accepted point of view - without feature modifier.
-                if (featureModifiers) {
-                    featureModifiers = featureModifiers.filter(
-                        fm => !(fm instanceof OmvPoliticalViewFeatureModifier)
-                    );
-                }
-                // If political view is indeed requested append feature modifier at the end of list.
-                if (politicalView.length !== 0) {
-                    assert(
-                        politicalView.length === 2,
-                        "The political view must be specified as two letters ISO 3166-1 standard!"
-                    );
-                    const povFeatureModifier = new OmvPoliticalViewFeatureModifier(politicalView);
-                    if (featureModifiers) {
-                        featureModifiers.push(povFeatureModifier);
-                    } else {
-                        featureModifiers = [povFeatureModifier];
-                    }
-                }
-                // Reset modifiers if nothing was added.
-                this.m_featureModifiers =
-                    featureModifiers && featureModifiers.length > 0 ? featureModifiers : undefined;
-            }
-
-            if (omvOptions.gatherFeatureAttributes !== undefined) {
-                this.m_gatherFeatureAttributes = omvOptions.gatherFeatureAttributes === true;
-            }
-            if (omvOptions.skipShortLabels !== undefined) {
-                this.m_skipShortLabels = omvOptions.skipShortLabels;
-            }
-            if (omvOptions.enableElevationOverlay !== undefined) {
-                this.m_enableElevationOverlay = omvOptions.enableElevationOverlay;
-            }
-            if (omvOptions.roundUpCoordinatesIfNeeded !== undefined) {
-                this.m_roundUpCoordinatesIfNeeded = omvOptions.roundUpCoordinatesIfNeeded;
-            }
-        }
-        if (options?.languages !== undefined) {
-            this.languages = options.languages;
-        }
+        this.m_options = {
```

Comment:
Minor, you could merge the `m_options` and `m_optionsChanged` into a class and then add a method update which would encapsulate the `m_optionsChanged = true` change.

Reply:
done.